src/python/WMComponent/DBS3Buffer/DBSBufferFile.py

```diff
@@ -387,8 +387,16 @@ def setProcessingVer(self, ver):
 
         Set the era
         """
-
-        self['processingVer'] = ver
+        if isinstance(ver, str) or isinstance(ver, bytes):
```

Comment:
I think I commented on this code in the other PR.
When it gets to this level, processing version has been extensively tested. Maybe the only thing that we would need to keep here is something like:
```
if ver is None:
    procVer = 0
```

in case this information isn't loaded from the database.

Reply:
I disagree since python does not provide a way to check input data-type. I don't rely on your word that provided input is and will be extensively tested. By definition someone can simply pass a string to this code just calling this method. And, I don't understand why you are so resistant against checking provided data-type. I discovered exactly this issue that somewhere in WMCore someone passed a wrong data-type and I'm not convinced that changing schema will prevent from doing this again form any other place in a WMCore code. I think **IT IS SAFE** to check explicitly for correct data type and only allow the one which is expected. The code does not add anything new and only provide additional checks.

Reply:
Yes, I know it's safer. I am just trying to find a good balance between being safe and keeping code as simple as possible.

I don't think it's impossible to build a DBSBufferFile/DBSBufferBlock without relying on the DAO information, even though WMAgents do so based on DAOs. But okay, let's keep these extra checks here then. I would like to suggest a different approach though, which relies less on data type checking (I'd avoid using anything that checks for bytes/str, as it gave us a nightmare over the past years).

Here is a suggestion for this code:
``` 
ver = ver or 0
try:
    ver = int(ver)
except TypeError:
    msg = "Provided procVer=%s of type %s cannot be converted to int" % (ver, type(ver))
    raise TypeError(msg)
```

or if we do not want to "massage" the exception message, we could remove the try/except as well.
What do you think? Could you please update this code accordingly?